compiler/src/iree/compiler/Codegen/Common/IREEComprehensiveBufferizePass.cpp

```diff
@@ -88,34 +116,11 @@ class IREEComprehensiveBufferizePass
   void runOnOperation() override;
 
 private:
-  const std::optional<BufferizationOptions::AllocationFn> allocationFn;
-  const std::optional<BufferizationOptions::MemCpyFn> memCpyFn;
+  const BufferizationOptions::AllocationFn allocationFn = defaultAllocationFn;
+  const BufferizationOptions::MemCpyFn memCpyFn = defaultMemCpyFn;
```

Comment:
This is the non-trivial part for `IREEComprehensiveBufferizePass` that mentioned in the PR description. @MaheshRavishankar please take a look.

Reply:
Actually I am confused what happens here. When is the default picked up ? Also the naming makes things hard to follow. Can you make the names different (from just `allocationFn` everywhere)